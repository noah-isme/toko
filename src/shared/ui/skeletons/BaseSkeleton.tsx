import * as React from 'react';

import { cn } from '@/lib/utils';

export interface BaseSkeletonProps {
  className?: string;
}

export function BaseSkeleton({ className }: BaseSkeletonProps) {
  return (
    <React.Fragment>
      <div
        aria-hidden="true"
        className={cn(
          'pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50',
          className,
        )}
      />
    </React.Fragment>
  );
}
