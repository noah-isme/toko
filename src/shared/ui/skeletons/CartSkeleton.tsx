import * as React from 'react';

import { srOnly } from '../a11y';

import { BaseSkeleton } from './BaseSkeleton';

export interface CartSkeletonProps {
  items?: number;
}

export function CartSkeleton({ items = 4 }: CartSkeletonProps) {
  return (
    <React.Fragment>
      <div className="space-y-4" role="status">
        {srOnly('Memuat...')}
        {Array.from({ length: items }).map((_, index) => (
          <div
            key={index}
            className="flex items-center gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
          >
            <BaseSkeleton className="h-16 w-16 rounded-md" />
            <div className="flex-1 space-y-2">
              <BaseSkeleton className="h-4 w-2/3" />
              <BaseSkeleton className="h-4 w-1/2" />
            </div>
            <BaseSkeleton className="h-10 w-16" />
          </div>
        ))}
        <div className="flex justify-end">
          <BaseSkeleton className="h-12 w-48" />
        </div>
      </div>
    </React.Fragment>
  );
}
