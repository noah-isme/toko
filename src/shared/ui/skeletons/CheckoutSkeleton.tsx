import * as React from 'react';

import { srOnly } from '../a11y';

import { BaseSkeleton } from './BaseSkeleton';

export function CheckoutSkeleton() {
  return (
    <React.Fragment>
      <div className="grid gap-8 lg:grid-cols-[minmax(0,2fr)_minmax(0,1fr)]" role="status">
        {srOnly('Memuat...')}
        <div className="space-y-6 rounded-lg border border-border/60 bg-muted/20 p-6">
          <div className="space-y-4">
            <BaseSkeleton className="h-6 w-1/3" />
            <div className="space-y-3">
              {Array.from({ length: 4 }).map((_, index) => (
                <BaseSkeleton key={index} className="h-10 w-full" />
              ))}
            </div>
          </div>
          <div className="space-y-4">
            <BaseSkeleton className="h-6 w-1/3" />
            <div className="space-y-3">
              {Array.from({ length: 3 }).map((_, index) => (
                <BaseSkeleton key={index} className="h-10 w-full" />
              ))}
            </div>
          </div>
          <BaseSkeleton className="h-12 w-48" />
        </div>
        <div className="space-y-4 rounded-lg border border-border/60 bg-muted/20 p-6">
          <BaseSkeleton className="h-6 w-1/2" />
          {Array.from({ length: 4 }).map((_, index) => (
            <BaseSkeleton key={index} className="h-4 w-full" />
          ))}
          <div className="space-y-2 pt-2">
            <BaseSkeleton className="h-4 w-2/3" />
            <BaseSkeleton className="h-10 w-full" />
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}
