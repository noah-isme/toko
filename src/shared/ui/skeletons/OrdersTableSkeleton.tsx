import * as React from 'react';

import { srOnly } from '../a11y';

import { BaseSkeleton } from './BaseSkeleton';

export interface OrdersTableSkeletonProps {
  rows?: number;
}

export function OrdersTableSkeleton({ rows = 8 }: OrdersTableSkeletonProps) {
  return (
    <React.Fragment>
      <div className="overflow-hidden rounded-lg border" role="status">
        {srOnly('Memuat...')}
        <table className="min-w-full divide-y divide-border/60">
          <thead className="bg-muted/30">
            <tr>
              {Array.from({ length: 4 }).map((_, index) => (
                <th
                  key={index}
                  className="px-4 py-3 text-left text-sm font-medium text-muted-foreground"
                >
                  <BaseSkeleton className="h-4 w-20" />
                </th>
              ))}
            </tr>
          </thead>
          <tbody className="divide-y divide-border/60 bg-background">
            {Array.from({ length: rows }).map((_, rowIndex) => (
              <tr key={rowIndex} className="bg-background">
                {Array.from({ length: 4 }).map((_, colIndex) => (
                  <td key={colIndex} className="px-4 py-3">
                    <BaseSkeleton className="h-4 w-24" />
                  </td>
                ))}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </React.Fragment>
  );
}
