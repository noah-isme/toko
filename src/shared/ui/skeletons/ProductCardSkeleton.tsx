import * as React from 'react';

import { srOnly } from '../a11y';

import { BaseSkeleton } from './BaseSkeleton';

export interface ProductCardSkeletonProps {
  count?: number;
}

export function ProductCardSkeleton({ count = 8 }: ProductCardSkeletonProps) {
  return (
    <React.Fragment>
      <div className="grid gap-6 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4" role="status">
        {srOnly('Memuat...')}
        {Array.from({ length: count }).map((_, index) => (
          <div
            key={index}
            className="flex flex-col gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
          >
            <BaseSkeleton className="aspect-video w-full" />
            <div className="space-y-2">
              <BaseSkeleton className="h-4 w-3/4" />
              <BaseSkeleton className="h-4 w-1/2" />
            </div>
            <BaseSkeleton className="mt-auto h-10 w-full" />
          </div>
        ))}
      </div>
    </React.Fragment>
  );
}
