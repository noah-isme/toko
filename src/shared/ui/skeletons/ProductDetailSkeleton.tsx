import * as React from 'react';

import { srOnly } from '../a11y';

import { BaseSkeleton } from './BaseSkeleton';

export function ProductDetailSkeleton() {
  return (
    <React.Fragment>
      <div className="grid gap-10 lg:grid-cols-[minmax(0,1.1fr)_minmax(0,1fr)]" role="status">
        {srOnly('Memuat...')}
        <div className="flex items-center justify-center rounded-lg border border-border/60 bg-muted/20 p-6">
          <BaseSkeleton className="aspect-square w-full max-w-xl" />
        </div>
        <div className="flex flex-col gap-6">
          <div className="space-y-3">
            <BaseSkeleton className="h-8 w-3/4" />
            <BaseSkeleton className="h-6 w-1/3" />
          </div>
          <div className="space-y-2">
            <BaseSkeleton className="h-4 w-full" />
            <BaseSkeleton className="h-4 w-5/6" />
            <BaseSkeleton className="h-4 w-2/3" />
          </div>
          <div className="space-y-2">
            <BaseSkeleton className="h-4 w-1/2" />
            <BaseSkeleton className="h-4 w-1/3" />
          </div>
          <div className="flex flex-col gap-4 sm:flex-row">
            <BaseSkeleton className="h-12 w-full sm:w-40" />
            <BaseSkeleton className="h-12 w-full sm:w-40" />
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}
