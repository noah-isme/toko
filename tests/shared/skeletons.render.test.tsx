import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import { afterEach, describe, expect, it } from 'vitest';

import { CartSkeleton } from '@/shared/ui/skeletons/CartSkeleton';
import { CheckoutSkeleton } from '@/shared/ui/skeletons/CheckoutSkeleton';
import { OrdersTableSkeleton } from '@/shared/ui/skeletons/OrdersTableSkeleton';
import { ProductCardSkeleton } from '@/shared/ui/skeletons/ProductCardSkeleton';
import { ProductDetailSkeleton } from '@/shared/ui/skeletons/ProductDetailSkeleton';

type SkeletonCase = {
  name: string;
  renderCase: () => ReturnType<typeof render>;
};

const skeletonCases: SkeletonCase[] = [
  {
    name: 'ProductCardSkeleton',
    renderCase: () => render(<ProductCardSkeleton count={2} />),
  },
  {
    name: 'ProductDetailSkeleton',
    renderCase: () => render(<ProductDetailSkeleton />),
  },
  {
    name: 'CartSkeleton',
    renderCase: () => render(<CartSkeleton items={3} />),
  },
  {
    name: 'CheckoutSkeleton',
    renderCase: () => render(<CheckoutSkeleton />),
  },
  {
    name: 'OrdersTableSkeleton',
    renderCase: () => render(<OrdersTableSkeleton rows={4} />),
  },
];

describe('Skeleton accessibility', () => {
  afterEach(() => {
    cleanup();
  });

  it.each(skeletonCases)('%s renders loading role with sr-only text', ({ renderCase }) => {
    const { container } = renderCase();

    expect(screen.getByRole('status')).toBeInTheDocument();
    expect(screen.getByText('Memuat...')).toHaveClass('sr-only');
    expect(container.firstChild).toMatchSnapshot();
  });
});
