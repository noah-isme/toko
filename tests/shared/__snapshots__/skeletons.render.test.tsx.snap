// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton accessibility > { name: 'CartSkeleton', renderCase: [Function renderCase] } renders loading role with sr-only text 1`] = `
<div
  class="space-y-4"
  role="status"
>
  <span
    class="sr-only"
  >
    Memuat...
  </span>
  <div
    class="flex items-center gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim bg-muted/50 h-16 w-16 rounded-md"
    />
    <div
      class="flex-1 space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-2/3"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/2"
      />
    </div>
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-16"
    />
  </div>
  <div
    class="flex items-center gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim bg-muted/50 h-16 w-16 rounded-md"
    />
    <div
      class="flex-1 space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-2/3"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/2"
      />
    </div>
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-16"
    />
  </div>
  <div
    class="flex items-center gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim bg-muted/50 h-16 w-16 rounded-md"
    />
    <div
      class="flex-1 space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-2/3"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/2"
      />
    </div>
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-16"
    />
  </div>
  <div
    class="flex justify-end"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-12 w-48"
    />
  </div>
</div>
`;

exports[`Skeleton accessibility > { name: 'CheckoutSkeleton', renderCase: [Function renderCase] } renders loading role with sr-only text 1`] = `
<div
  class="grid gap-8 lg:grid-cols-[minmax(0,2fr)_minmax(0,1fr)]"
  role="status"
>
  <span
    class="sr-only"
  >
    Memuat...
  </span>
  <div
    class="space-y-6 rounded-lg border border-border/60 bg-muted/20 p-6"
  >
    <div
      class="space-y-4"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-6 w-1/3"
      />
      <div
        class="space-y-3"
      >
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
      </div>
    </div>
    <div
      class="space-y-4"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-6 w-1/3"
      />
      <div
        class="space-y-3"
      >
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
        <div
          aria-hidden="true"
          class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-12 w-48"
    />
  </div>
  <div
    class="space-y-4 rounded-lg border border-border/60 bg-muted/20 p-6"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-6 w-1/2"
    />
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-full"
    />
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-full"
    />
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-full"
    />
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-full"
    />
    <div
      class="space-y-2 pt-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-2/3"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-10 w-full"
      />
    </div>
  </div>
</div>
`;

exports[`Skeleton accessibility > { name: 'OrdersTableSkeleton', renderCase: [Function renderCase] } renders loading role with sr-only text 1`] = `
<div
  class="overflow-hidden rounded-lg border"
  role="status"
>
  <span
    class="sr-only"
  >
    Memuat...
  </span>
  <table
    class="min-w-full divide-y divide-border/60"
  >
    <thead
      class="bg-muted/30"
    >
      <tr>
        <th
          class="px-4 py-3 text-left text-sm font-medium text-muted-foreground"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-20"
          />
        </th>
        <th
          class="px-4 py-3 text-left text-sm font-medium text-muted-foreground"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-20"
          />
        </th>
        <th
          class="px-4 py-3 text-left text-sm font-medium text-muted-foreground"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-20"
          />
        </th>
        <th
          class="px-4 py-3 text-left text-sm font-medium text-muted-foreground"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-20"
          />
        </th>
      </tr>
    </thead>
    <tbody
      class="divide-y divide-border/60 bg-background"
    >
      <tr
        class="bg-background"
      >
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
      </tr>
      <tr
        class="bg-background"
      >
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
      </tr>
      <tr
        class="bg-background"
      >
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
      </tr>
      <tr
        class="bg-background"
      >
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
        <td
          class="px-4 py-3"
        >
          <div
            aria-hidden="true"
            class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-24"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Skeleton accessibility > { name: 'ProductCardSkeleton', renderCase: [Function renderCase] } renders loading role with sr-only text 1`] = `
<div
  class="grid gap-6 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
  role="status"
>
  <span
    class="sr-only"
  >
    Memuat...
  </span>
  <div
    class="flex flex-col gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 aspect-video w-full"
    />
    <div
      class="space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-3/4"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/2"
      />
    </div>
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 mt-auto h-10 w-full"
    />
  </div>
  <div
    class="flex flex-col gap-4 rounded-lg border border-border/60 bg-muted/20 p-4"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 aspect-video w-full"
    />
    <div
      class="space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-3/4"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/2"
      />
    </div>
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 mt-auto h-10 w-full"
    />
  </div>
</div>
`;

exports[`Skeleton accessibility > { name: 'ProductDetailSkeleton', renderCase: [Function renderCase] } renders loading role with sr-only text 1`] = `
<div
  class="grid gap-10 lg:grid-cols-[minmax(0,1.1fr)_minmax(0,1fr)]"
  role="status"
>
  <span
    class="sr-only"
  >
    Memuat...
  </span>
  <div
    class="flex items-center justify-center rounded-lg border border-border/60 bg-muted/20 p-6"
  >
    <div
      aria-hidden="true"
      class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 aspect-square w-full max-w-xl"
    />
  </div>
  <div
    class="flex flex-col gap-6"
  >
    <div
      class="space-y-3"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-8 w-3/4"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-6 w-1/3"
      />
    </div>
    <div
      class="space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-full"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-5/6"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-2/3"
      />
    </div>
    <div
      class="space-y-2"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/2"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-4 w-1/3"
      />
    </div>
    <div
      class="flex flex-col gap-4 sm:flex-row"
    >
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-12 w-full sm:w-40"
      />
      <div
        aria-hidden="true"
        class="pointer-events-none animate-pulse prm:no-anim rounded bg-muted/50 h-12 w-full sm:w-40"
      />
    </div>
  </div>
</div>
`;
